### PR TITLE
doc: added the new 'init_ttl' argument to the shdict:incr() documentation.

### PR DESCRIPTION
**Pull request by thibaultcha**
I hereby granted the copyright of the changes in this pull request
to the authors of this lua-nginx-module project.

### CODE REVIEW

File: README.markdown [hunk 1]
```diff
@@ -6464,6 +6464,8 @@ When the key does not exist or has already expired in the shared dictionary,
 
 Like the [add](#ngxshareddictadd) method, it also overrides the (least recently used) unexpired items in the store when running out of storage in the shared memory zone.
 
+The optional `init_ttl` argument specifies expiration time (in seconds) of the value when it is initialized via the `init` argument. The time resolution is `0.001` seconds. If `init_ttl` takes the value `0` (which is the default), then the item will never expire. This argument cannot be provided without providing the `init` argument as well, and has no effect if the value already exists (e.g., if it was previously inserted via [set](#ngxshareddictset) or the likes).
```
Comment (thibaultcha) [inline README.markdown]: Missing the requirement upon the `resty.core` module - will update.

Comment (thibaultcha) [reply README.markdown]: Addressed

File: README.markdown [hunk 2]
```diff
@@ -6464,6 +6466,25 @@ When the key does not exist or has already expired in the shared dictionary,
...
+ local cats = ngx.shared.cats
+ local newval, err = cats:incr("black_cats", 1, 0, 0.1)
+
+ print(newval) -- 1
+
+ ngx.sleep(0.2)
+
+ local val, err = cats:get("black_cats")
+ print(val) -- nil
+```
```
Comment (thibaultcha) [inline README.markdown]: Unsure if this example is desired

**Comment by thibaultcha** [comment]
@agentzh Just a ping that if we are about to tag a release for ngx_lua, we should probably include this documentation in it. (there is the bugfix for ttls too, but up to you.)

File: README.markdown [hunk 3]
```diff
@@ -6476,6 +6497,8 @@ This method was first introduced in the `v0.3.1rc22` release.
 
 The optional `init` parameter was first added in the `v0.10.6` release.
 
+The optional `init_ttl` parameter was introduced in the `v0.10.12rc2` release.
```
Comment (thibaultcha) [inline README.markdown]: Updated to 0.10.12rc2

**Comment by agentzh** [comment]
@thibaultcha Merged. Thanks!

**Pull request closed without merging**